feat: Checkbox label ellipsis overflow

salarenko opened the pull request:
## Related Issue
Closes SAP/fundamental-styles#610

## Description
In previous implementation text inside `.fd-checkbox__label` was not able to properly overflow with ellipsis, because of `display: flex`;

This PR introduces new wrapper which should be used inside `.fd-checkbox__label` to provide proper overflow for the text values.

#### Please check whether the PR fulfills the following requirements

- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-styles/wiki/PR-Review-Checklist

meganaconley commented:
A11y looks good - full label is read out when truncated. Seconding request for an example but otherwise looks good.

salarenko commented:
@InnaAtanasova I've also noticed this behavior in storybbok examples, but this is not related to checkbox itself.
Checkbox examples are grouped in `fieldset` and fieldsets are displayed in flex container. None of flexbox examples wrappers up to `class="root"` container has set width limitations, that's why example is stretching as much as it wants to, no matter if checkbox text/labels have some percentage based width limitation defined.

`.fd-checkbox__label` has `max-width: 100%` limitation. I could at it to `.fd-checkbox__text` but I don't think this is necessary because there is no other way of using `.fd-checkbox__text` than putting it inside `.fd-checkbox__label`.

This is how it works if you remove `display: flex` or define `max-width` from wrapping containers:
![checkbox-overflow](https://user-images.githubusercontent.com/17496353/86894424-539bbf00-c103-11ea-893e-1b920974e29c.gif)

salarenko commented:
@InnaAtanasova, @meganaconley
- I've updated sorybook styling, so now you can easily reproduce wrapping behavior
- I've also updated Breaking Changes
